### PR TITLE
Acceptance testing accounts kev

### PR DESCRIPTION
**Pull request by KevinMonahanUWM**
All acceptance tests might be wrong because I unsure of how sessions impact testing. If it is not an issue then they are probably okay but let me know if I need more tests or something looks wrong to you. Also I deleted the Templates directory in this commit to remove the unneeded directory.

### CODE REVIEW

**Review by JosiahDeYong — APPROVED** [review]
Looks good

File: TAScheduler/AcceptanceTests/test_create_account.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        self.account = Administrator(user=temp)
+        self.account.save()
+        self.info = {"email_address", "test@uwm.edu", "password", "pass", "first_name", "test", "last_name", "ignore",
+                     "home_address", "3400 N Maryland Ave", "phone_number", 4142292222, "role", "TA"}
+
+    # TODO need to make tests work with sessions but I am unsure of how to start that
+    def test_creation(self):
+        self.user.post("/home/manageaccount/create", self.info)
+        self.assertTrue(User.objects.filter(email_address="test@uwm.edu").count() is 1)
+
```
Comment (Kiran-Ee) [inline TAScheduler/AcceptanceTests/test_create_account.py:25]: Maybe another test: even though that object is in the database, does it contain all of the info we sent into it? 
Also, maybe another test for a .context["message"] = "successfully created account."

File: TAScheduler/AcceptanceTests/test_create_account.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        self.account.save()
+        self.info = {"email_address", "testadmin@uwm.edu", "password", "pass", "first_name", "test", "last_name",
+                     "ignore", "home_address", "3400 N Maryland Ave", "phone_number", 4142292222, "role", "TA"}
+
+    # TODO need to make tests work with sessions but I am unsure of how to start that
+    def test_existing_user(self):
+        resp = self.user.post("/home/manageaccount/create", self.info)
+        self.assertEquals(resp.context["message"], "Duplicate User",
+                          "You created an account that already exists")
+
```
Comment (Kiran-Ee) [inline TAScheduler/AcceptanceTests/test_create_account.py:47]: Add another method to ensure that the database is not actually change. For example, even though the context might say "Duplicate User", it's possible that it still adds it to the database. Solution: maybe check the # of user in the db before and after the post to ensure nothings changed

File: TAScheduler/AcceptanceTests/test_create_account.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        self.assertEquals(resp.context["message"], "Missing Information",
+                          "You created an account with not all fields filled out")
+
+    def test_six_input(self):
+        resp = self.user.post("/home/manageaccount/create",
+                              {"password", "pass", "first_name", "test", "last_name", "ignore",
+                               "home_address", "3400 N Maryland Ave", "phone_number", 4142292222, "role", "TA"})
+        self.assertEquals(resp.context["message"], "Missing Information",
+                          "You created an account with not all fields filled out")
+
```
Comment (Kiran-Ee) [inline TAScheduler/AcceptanceTests/test_create_account.py:76]: Probably going to need a test for each field that could be missing- UNLESS, there's a better way I'm not aware (IF you figure out a better way to test all the permutations, could you let me know?)

File: TAScheduler/AcceptanceTests/test_create_account.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        self.user = Client()
+        temp = User(email_address="testadmin@uwm.edu", password="pass", first_name="Test", last_name="Test",
+                    home_address="Random location", phone_number=9990009999)
+        temp.save()
+        self.account = Administrator(user=temp)
+        self.account.save()
+        self.info = {"email_address", "test@uwm.edu", "password", "pass", "first_name", "test", "last_name", "ignore",
+                     "home_address", "3400 N Maryland Ave", "phone_number", 4, "role", "TA"}
+
+    # TODO need to make tests work with sessions but I am unsure of how to start that
```
Comment (Kiran-Ee) [inline TAScheduler/AcceptanceTests/test_create_account.py:93]: I was debating on whether to allow the admin to edit the "Unique ID" ... I would say no, but if you think we should allow this (please let me know) then there could be another test for that here.

File: TAScheduler/AcceptanceTests/test_delete_account.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        self.account = Administrator(user=temp)
+        self.account.save()
+        self.tempUser = User(email_address="test@uwm.edu", password="pass", first_name="test", last_name="ignore",
+                    home_address="3400 N Maryland Ave", phone_number=4142292222)
+        self.tempUser.save()
+        TA(user=self.tempUser, grader_status=False).save()
+
+    def test_correct_delete(self):
+        self.user.post("/home/manageaccount/delete", {"selection", self.tempUser})
+        self.assertNotIn(self.tempUser, User.objects, "Did not successfully delete user")
```
Comment (Kiran-Ee) [inline TAScheduler/AcceptanceTests/test_delete_account.py:26]: add a test for confirmation message.

File: TAScheduler/AcceptanceTests/test_edit_account.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        self.account.save()
+        self.tempUser = User(email_address="test@uwm.edu", password="pass", first_name="test", last_name="ignore",
+                             home_address="3400 N Maryland Ave", phone_number=4142292222)
+        self.tempUser.save()
+        TA(user=self.tempUser, grader_status=False).save()
+
+    def test_success_change(self):
+        self.user.post("/home/manageaccount/edit", {"User", self.tempUser}, follow=True)
+        self.user.post("/home/manageaccount/edit", {"first_name", "Paul"})
+        self.assertEquals("Paul", User.objects.get(self.tempUser), "Did not successfully edit account")
```
Comment (Kiran-Ee) [inline TAScheduler/AcceptanceTests/test_edit_account.py:27]: add confirmatoin message.

File: TAScheduler/AcceptanceTests/test_edit_account.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        temp.save()
+        self.account = Administrator(user=temp)
+        self.account.save()
+        self.tempUser = User(email_address="test@uwm.edu", password="pass", first_name="test", last_name="ignore",
+                             home_address="3400 N Maryland Ave", phone_number=4142292222)
+        self.tempUser.save()
+        TA(user=self.tempUser, grader_status=False).save()
+
+    def test_formatting_error(self):
+        self.user.post("/home/manageaccount/edit", {"User", self.tempUser}, follow=True)
```
Comment (Kiran-Ee) [inline TAScheduler/AcceptanceTests/test_edit_account.py:49]: add a test to ensure it didn't actually edit the account

File: TAScheduler/AcceptanceTests/test_edit_account.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        self.account = Administrator(user=temp)
+        self.account.save()
+
+    def test_discard(self):
+        resp = self.user.post('/home/manageaccount/edit', {"discard", True}, follow=True)
+        self.assertRedirects(resp, "/home/manageaccount/edit", status_code=302,
+                             target_status_code=200,
+                             fetch_redirect_response=True,
+                             msg="Discard changes did not redirect back to edit main page")
+        self.assertIn(self.account.user, User.objects, "User was changed when it should still be the same")
```
Comment (Kiran-Ee) [inline TAScheduler/AcceptanceTests/test_edit_account.py:76]: Couldn' the user "be in" the User.objects, but still be changed? I am looking at this tests and it seems that it just checks if the account exists but we would want to see if the fields have changed. (Unless .assertIn automatically accounts for this?)

**Review by Kiran-Ee — COMMENTED** [review]
Just a few suggestions and potential method ideas
- Your code is much cleaner than mine so I will steal a few of your formatting things after everyone has reviewed my code

File: TAScheduler/AcceptanceTests/test_create_account.py (hunk 4 above)
Comment (KevinMonahanUWM) [reply TAScheduler/AcceptanceTests/test_create_account.py:93]: No we can not modify the unique ID it is private to the user object and should not be modified. That is not something we should be messing with.

File: TAScheduler/AcceptanceTests/test_edit_account.py (hunk 8 above)
Comment (KevinMonahanUWM) [reply TAScheduler/AcceptanceTests/test_edit_account.py:76]: My understanding of this is that my user should not change and if the user did get changed when it should have changed the user object I have will not match the user stored in the database. But I could be wrong

**Review by AlecBrookens — APPROVED** [review]

**Review by Randall-S4 — COMMENTED** [review]
looks clean to me I do agree with what others are saying

**Pull request merged**